<launch>
  <!-- only SE2 is supported -->
  <arg name="sampling_space" value="SE2" />

  <node pkg="differentiable_rmap" type="NodeRmapPlanningFootstep" name="rmap_planning_footstep"
        output="screen">
    <rosparam subst_value="true">
      sampling_space: $(arg sampling_space)
      config_path: $(find differentiable_rmap)/config/RmapPlanningFootstep.yaml
      svm_path: /tmp/rmap_svm_model_$(arg sampling_space).libsvm
      bag_path: /tmp/rmap_grid_set_$(arg sampling_space).bag
    </rosparam>
  </node>

  <node pkg="optmotiongen" type="InteractiveMarkerManager.py" name="rviz_server"
        clear_params="true">
    <rosparam>
      interactive_markers:
        - name: target
          initial_pose:
            pos: [0, 0, 0.1]
          controls: [tx, ty, rz]
          markers:
            - type: 2 # SPHERE
              scale: [0.15, 0.15, 0.15]
              color: [1.0, 1.0, 0.0, 0.5]
            - type: 0 # ARROW
              scale: [0.15, 0.05, 0.05]
              color: [1.0, 1.0, 0.0, 0.5]
    </rosparam>
  </node>

  <node pkg="rviz" type="rviz" name="rviz"
        args="-d $(find differentiable_rmap)/rviz/RmapPlanningFootstep.rviz"
        output="log" clear_params="true" />
</launch>
